src/Microsoft.ML.Data/Data/SchemaDefinition.cs

```diff
@@ -32,7 +32,9 @@ public KeyTypeAttribute()
         }
 
         /// <summary>
-        /// Marks member as <see cref="KeyDataViewType"/> and specifies <see cref="KeyDataViewType"/> cardinality.
+        /// Marks member as <see cref="KeyDataViewType"/>. The <paramref name="count"/> should be set to
+        /// one more than the maximum value for the keys (to account for missing values).
+        /// If the values are outside of the specified cardinality they will be mapped to the missing value representation: 0.
         /// </summary>
```

Comment:
I think that the old description was correct, especially when the underlying values are not numbers. When that's the case, I think it might make more sense to think about cardinality instead of maximum value of keys.

Reply:
Maybe it could be added as a note for the special case of integer values?

---
In reply to: [273266746](https://github.com/dotnet/machinelearning/pull/3211#discussion_r273266746) [](ancestors = 273266746)